src/core/smp.cc

```diff
@@ -42,7 +42,7 @@ struct smp_service_group_impl {
 #endif
 };
 
-static smp_service_group_semaphore smp_service_group_management_sem{1, named_semaphore_exception_factory{"smp_service_group_management_sem"}};
+static thread_local smp_service_group_semaphore smp_service_group_management_sem{1, named_semaphore_exception_factory{"smp_service_group_management_sem"}};
```

Comment:
If it is accessed only on shard 0 how it would seem that it's being accessed across shards
in debug mode? Isn't main thread becomes shard 0?

Reply:
the only users of `smp_service_group_management_sem` are `create_smp_service_group()` and `destroy_smp_service_group()`. both of them can be called on non-shard 0, and they schedule the continuation on shard 0. i am not sure what do you mean by "it is accessed across shard"?

Reply:
I am referring to the commit message.

Reply:
My guess is that https://github.com/scylladb/seastar/blob/c0bf5768337dcec5e764700a4bcade33cca731e3/include/seastar/core/shared_ptr_debug_helper.hh#L40
doesn't work yet during static (in contrast to thread_local) initialization.